SRPG-44 gacha

SeRDic0705 opened the pull request:
## 개요
### 가챠기능 구현을 위한 GachaManager를 추가했습니다.
가챠매니저는 다음과 같은 로직으로 동작합니다.
- UI에서 가챠실행버튼 클릭시, Draw() 메서드는 tableName을 넘겨받습니다.
- tableName을 캐시에서 참조, 없다면 DB에서 로드 후 캐시에 저장합니다.
- 참조한 확률테이블 내에서 캐릭터 id를 하나 뽑아냅니다.
- 이미 보유중인 캐릭터인지를 확인하여 획득 처리 후, 가챠 연출을 위해 UI로 캐릭터 id를 반환합니다. (10연차를 진행했다면 캐릭터 id들을 queue로 전달합니다.

### AccountData에 VersionData를 추가했습니다.
- VersionData는 현재 게임의 버전, 진행중인 가챠정보, 진행중인 이벤트정보, 천장포인트등을 포함합니다.
- 게임 실행 후, 매 로그인 시마다 서버의 버전정보와 대조하여 다르다면 데이터를 새로 다운로드합니다.

### 가챠기능의 실행을 위한 GachaUI의 초안을 작성했습니다.
가챠 화면에 필요한 버튼과 이미지들은 구성했으나, 다른 파트의 기능구현이 필요한 부분이 많아 로직은 아직 생성하지 못했습니다

## PR 유형
어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다. (ex. feature/SRPG-11-UIBase)
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

SeRDic0705 committed:
build: Add VsCode Extention

SeRDic0705 committed:
feat: Constants에 GachaType추가

통상인지 픽업인지 가챠의 타입을 나타내기 위한 열거형 추가

#SRPG-44

SeRDic0705 committed:
feat: AccountData에 gachaPoint추가

픽업가챠의 천장포인트를 저장하기 위한 gachaPoint 추가

#SRGP-44

SeRDic0705 committed:
feat: 가챠 로직 구현

UI에서 GachaType과 tablename을 넘겨받아 실제로 실행하는 GachaManager 로직 구현

#SRPG-44

SeRDic0705 committed:
feat: Manager에 GachaManager 추가

UI에서 가챠 로직을 실행하기 위한 manager 추가

#SRPG-44

SeRDic0705 committed:
Merge branch 'develop' into SRPG-44-Gacha

SeRDic0705 committed:
feat: VersionData의 추가

- 현재 게임의 버전, 진행중인 가챠, 진행중인 이벤트 등의 정보를 가진 버전데이터를 추가했습니다.
- 게임 실행시마다 서버의 버전과 비교하여 동기화하는 로직이 필요합니다

#SRPG-46
#SRPG-44
#SRPG-29

SeRDic0705 committed:
feat: 가챠 UI 초안 추가

- 가챠를 위한 UI 클래스입니다.
- 픽업배너 클릭 시, gachaType과 tableName을 변경해야합니다.
- 재화가 충분한지 비교하는 로직의 추가가 필요합니다

#SRPG-44
#SRPG-16
#SRPG-69
#SRPG-68

SeRDic0705 committed:
fix: 버그 수정

AccountData에서 버전데이터를 리팩터링하며 생긴 버그 수정

#SRPG-44

SeRDic0705 committed:
Merge branch 'develop' into SRPG-44-Gacha

SeRDic0705 committed:
chore: 주석 깨짐 수정

GeCrabKani reviewed (changes requested):
GachaManager가 MonoBehaviour를 상속받지 않아도 될 것 같습니다.

GachaManager의 10회 뽑기에서도 가챠 포인트는 1만 올려주고 있습니다.

SeRDic0705 committed:
chore: GachaManager 수정

- 더이상 MonoBehaviour를 상속하지 않습니다.
- 2성 이상 테이블의 네임을 더 직관적으로 바꿨습니다
- 10연차에서 포인트가 1이 오르던 것을 정상값으로 수정했습니다

#SRPG-44

SeRDic0705 committed:
refactor: gachaPoint

- 마이너 업데이트의 경우 버전정보가 바뀌어도 천장포인트는 유지되어야 하기 때문에, gachaPoint를 AccountData에서 직접 접근하도록 다시 수정했습니다

#SRPG-44
#SRPG-46
#SRPG-29
